### PR TITLE
[Framework] Add dependencies check for code object freezing

### PR DESCRIPTION
**Pull request by movekevin**
## Type of Change
- [ ] New feature
- [ x ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ x ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
E2E tests

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/code.move [hunk 1]
```diff
@@ -214,10 +214,18 @@ module aptos_framework::code {
...
             let package: &mut PackageMetadata = pack;
             package.upgrade_policy = upgrade_policy_immutable();
         });
+
+        // We unfortunately have to make a copy of each package to avoid borrow checker issues as check_dependencies
+        // needs to borrow PackageRegistry from the dependency packages.
+        // This would increase the amount of gas used, but this is a rare operation and it's rare to have many packages
+        // in a single code object.
+        vector::for_each(registry.packages, |pack| {
+            check_dependencies(code_object_addr, &pack);
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/code.move:227]: This `check_dependencies ` function returns `allowed_module_deps` that are normally used in the native bytecode dep verification (https://github.com/aptos-labs/aptos-core/blob/3b50ea6e4c3170cfaaa02193b0334cbabeaae875/aptos-move/framework/aptos-framework/sources/code.move#L266-L266). Is it fine to ignore the return value?

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/code.move:227]: Yeah here in this freeze function we only care about visibility checks as other checks should have already been performed when this package was first deployed or last updated

File: aptos-move/e2e-move-tests/src/tests/object_code_deployment.rs [hunk 2]
```diff
@@ -362,6 +373,37 @@ fn freeze_code_object_fail_when_not_owner() {
...
+        MUT_DEPS_MODULE_ADDRESS_NAME.to_string(),
+        context.object_address,
+    );
+    assert_success!(context.execute_object_code_action_with_options(
+        &acc,
+        "object_code_deployment.data/pack_mutable_deps",
+        ObjectCodeAction::Deploy,
+        options
+    ));
+    let status = context.execute_object_code_action(&acc, "", ObjectCodeAction::Freeze);
```
Comment (JohnChangUK) [inline aptos-move/e2e-move-tests/src/tests/object_code_deployment.rs:403]: nit: Would be nice to have a comment mentioning freezing a package with mutable dependencies is not allowed.

**Comment by JohnChangUK** [comment]
LGTM, maybe one more test to verify freezing an upgradeable/mutable package with immutable dependencies should work too.

**Review by junkil-park — APPROVED** [review]

**Review by davidiw — APPROVED** [review]

**Comment by movekevin** [comment]
> LGTM, maybe one more test to verify freezing an upgradeable/mutable package with immutable dependencies should work too.
> 
> Edit: Added this in!

Where did you add this? I don't see the additional test

**Pull request merged**